qiita_db/base.py

```diff
@@ -256,21 +256,20 @@ def check_status(self, status, exclude=False):
         Table setup:
         foo: foo_status_id  ----> foo_status: foo_status_id, status
         """
-        # Check that self._table is actually defined
-        self._check_subclass()
-
-        # Get all available statuses
-        conn_handler = SQLConnectionHandler()
-
-        statuses = [x[0] for x in conn_handler.execute_fetchall(
-            "SELECT DISTINCT status FROM qiita.{0}_status".format(self._table),
-            (self._id, ))]
-
-        # Check that all the provided statuses are valid statuses
-        if set(status).difference(statuses):
-            raise ValueError("%s are not valid status values"
-                             % set(status).difference(statuses))
-
-        # Get the DB status of the object
-        dbstatus = self.status
-        return dbstatus not in status if exclude else dbstatus in status
+        with TRN:
+            # Get all available statuses
+            sql = "SELECT DISTINCT status FROM qiita.{0}_status".format(
+                self._table)
+            TRN.add(sql, [self._id])
+            # We need to access to the results of the last SQL query,
+            # hence indexing using -1
+            avail_status = [x[0] for x in TRN.execute()[-1]]
```

Comment:
Uhm, the difference is that here is not returning a column, here is returning the results of the last query, which is a list of lists... An option might be `def execute_fetchindex(index=-1)` which defaults always to the last one...